Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -1923,6 +2006,16 @@ protected class ClassWriter : IClassWriter {
         return IdProtect(udt.CompileName);
       }
 
+      //Use the interface if applicable (not handwritten, or incompatible variance)
+      bool compileIt = true;
+      if ((cl is IndDatatypeDecl || cl is CoDatatypeDecl)
+          && !ignoreInterface
+          && (!Attributes.ContainsBool(cl.Attributes, "compile", ref compileIt) || compileIt)
```

Comment:
Why do we need to take into account `:compile` here?

Reply:
The handwritten material from `DafnyRuntime` doesn't respect our new structure for datatypes, so we either have to exclude it, or change it.

Reply:
Is that for datatypes? If so, then I think we should rewrite `DafnyRuntime` accordingly.